### ISSUE TITLE
Guild does not recognize short form argparse parameters

### ISSUE THREAD

**Issue by gsanmartinsilva** [guildai/guildai#112]
When using the argparse library to specify hyperparameters, guild does not recognize the short form, but it works correctly and as expected for the long form.

For example, when defining the learning rate and epochs:

```
parser.add_argument('--learning_rate', action="store", dest="l_r", type=int, default=1e-3)
parser.add_argument('-epochs', action="store", dest="epochs", type=int, default=500)
```

guild will correctly recognize learning rate, but not epochs.

**Comment by gar1t** [comment]
This should be fixed as an enhancement for Guild to make any claim to "no code change" support!

Note on implementation... Guild supports `arg-name`, which is a general purpose attribute that applies not just to argument 'names' but also to global variable names -- and possibly other topics depending on the value of `args-dest`.

Thoughts on ways users might configure this:

1. Support optional leading dashes in `arg-name` where a single dash signals the intended form
2. Attribute that indicates the form (e.g. `arg-form=short`)
3. Attribute that indicates the arg template (e.g. `arg-template='-epochs {}'`)
4. Expand `args-dest` to flag level and introduce another value `short-args` - this approach has the advantage of setting the default form at the operation level
